geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -414,6 +429,28 @@ private static double processByteArrayAsDouble(byte[] value) {
     }
   }
 
+  // Comparison to allow the use of bLEAST_MEMBER_NAME and bGREATEST_MEMBER_NAME to always be less
+  // than or greater than respectively any other byte array representing a member name. The use of
+  // "==" is important as it allows us to differentiate between the constant byte arrays defined
+  // in StringBytesGlossary and user-supplied member names which may be equal in content but have
+  // a different memory address.
+  private static int checkDummyMemberNames(byte[] array1, byte[] array2) {
```

Comment:
I'm a little concerned that this is on a hot path and that the primary code path will require traversing all the conditionals. One alternative I can think of is to have these special sentinels try and actually be min/max values - `byte[] {-128, -128, -128, ...}` and `byte[] {127, 127, 127, ...}` for some largish sized byte array. That way the comparisons can just happen naturally. I realize though that in an extreme corner case a user may actually try and add such a value. I'm OK leaving it like this for now, but at some point I'd like to satisfy myself on the performance aspects.

Reply:
That could potentially work for the least possible member name, which would just be `byte[] {-128}` but to have a true greatest possible member name, it'd have to be `byte[] {127, 127, ...}` for the longest possible byte array, as any byte array longer than it would evaluate to being greater than it. Using such a long byte array in the "normal" comparison method potentially leads to a lot more operations, since we have to compare element by element rather than just do one `==` operation. I see your point though, so I'll refactor this method to do fewer checks if neither of the arrays are equal to the constants. If it's any comfort, Redis uses the approach implemented here, where they have a defined constant as their min/man (they use "minstring" and "maxstring").